src/managers/RoleManager.js

```diff
@@ -143,6 +144,53 @@ class RoleManager extends BaseManager {
       });
   }
 
+  /**
+   * Edits a role of the guild.
+   * @param {RoleResolvable} role The role to edit
+   * @param {RoleData} data The new data for the role
+   * @param {string} [reason] Reason for editing this role
+   * @returns {Promise<Role>}
+   * @example
+   * // Edit a role
+   * guild.roles.edit('222079219327434752', { name: 'buddies' })
+   *   .then(updated => console.log(`Edited role name to ${updated.name}`))
+   *   .catch(console.error);
+   */
+  async edit(role, data, reason) {
+    const id = this.resolveID(role);
+    if (!id) throw new TypeError('INVALID_TYPE', 'role', 'RoleResolvable');
+
+    if (typeof data.position === 'number') {
+      const updatedRoles = await setPosition(
+        this,
```

Comment:
The first parameter here should be a role instance, otherwise this will just patch the roles without changing the position of the role in question.

Reply:
Nice catch! I'm unsure why this worked just fine when I tested it. Fixed in b1a5b52.